File: java/api/src/main/java/org/dressdiscover/api/models/user/UserBookmarkId.java
package org.dressdiscover.api.models.user;

import org.thryft.waf.api.models.IntegerModelId;

public final class UserBookmarkId extends IntegerModelId {
    public static UserBookmarkId parse(final String value) throws InvalidUserBookmarkIdException {
        try {
            return new UserBookmarkId(Integer.parseInt(value));
        } catch (final IllegalArgumentException e) {
            throw InvalidUserBookmarkIdException.create("not an integer");
        }
    }

    public UserBookmarkId(final int value) {
        super(value);
    }
}

File: thrift/src/dressdiscover/api/models/user/user_bookmark_id.py
import os.path, sys
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
from _id import _Id


class UserBookmarkId(_Id):
    def java_qname(self, **kwds):
        return 'org.dressdiscover.api.models.user.UserBookmarkId'

    def java_read_protocol_throws_checked(self):
        return ['org.dressdiscover.api.models.user.InvalidUserBookmarkIdException']
